.github/workflows/python-test.yml

```diff
@@ -0,0 +1,44 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python application
+
+on: [push, pull_request]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
+        # if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+        if [ -f test-requirements.txt ]; then \
+          pip install -r test-requirements.txt; \
+        elif [ -f requirements.txt ]; then \
+           pip install -r requirements.txt; \
+        fi
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
```

Comment:
Great point -- this was taken verbatim from Github's example action for python, but what you said makes a lot of sense.  The [default value appears to be 79](https://flake8.pycqa.org/en/latest/user/options.html#cmdoption-flake8-max-line-length), so I'll just remove it.

The second flake8 call looks like it just outputs violations as warnings, but the first flake8 call can actually fail the action (but only for selected rules: E9, F63, F7, and F82).  We could make it really strict and just get rid of the first call and remove the "--exit-zero" from the second call which will enforce all flake8 code conventions. 

I'll check to see what it would take to bring everything up to standards and modify this configuration to enforce them all

Reply:
@ScholliYT, ready for review!